Registry mvp docs (#689)
wandb/docs: docs/guides/registry/intro.md

---
slug: /guides/registry
displayed_sidebar: default
---

# Registry

:::info

> W&B Registry is in active development.  

Reviewer: ```suggestion
W&B Registry is in private preview. Please contact your account team or support@wandb.com for early access.  
```

:::

Use W&B Registry to share artifacts from your machine learning pipeline, such as machine learning and datasets, across teams within an organization. 

## How it works
W&B Registry is composed of three main components: registries, collections, and [artifact versions](../artifacts/create-a-new-artifact-version.md).

> A *registry* is a [INSERT]. You can think of a registry as the top most level of a directory. Each registry consists of one or more sub directories called collections. A *collection* is a folder or a set of linked [*artifact versions*](../artifacts/create-a-new-artifact-version.md) inside a registry. An [artifact version](../artifacts/create-a-new-artifact-version.md) is [INSERT].  A registry belongs to an organization, not a specific team.

Reviewer: ```suggestion
A *registry* is a repository or catalog for ML assets of the same kind. You can think of a registry as the top most level of a directory. Each registry consists of one or more sub directories called collections. A *collection* is a folder or a set of linked [*artifact versions*](../artifacts/create-a-new-artifact-version.md) inside a registry. An [artifact version](../artifacts/create-a-new-artifact-version.md) is a single, immutable snapshot of an artifact at a particular stage of its development.  A registry belongs to an organization, not a specific team.
```

Consider the following example, suppose your company wants to explore brand new cat classification models within your organization. To do this, you create a registry called "Cat Models registry". You are not sure which image classification is best, so you create multiple machine learning experiments with varying hyperparameters and algorithms. To organize your experiments and results, you create a collection for each model algorithm that experiment. Within that collection, you link the best model artifacts from your experiments to that collection.

> <!-- To do: Add Raven's new diagram -->

Reviewer: <img width="709" alt="Registry-structure-graphic-docs" src="https://github.com/wandb/docodile/assets/40642416/48a105c9-1911-4c3e-a770-5f66c7525d8c">

Raven has updated this so we should be good to go with inserting!

> ## Migrating from W&B Model Registry to W&B Registry

Reviewer: ```suggestion
## Migrating from team W&B Model Registry to W&B Registry
```

> [INSERT blurb on how this impacts customers]

Reviewer: ```suggestion
If your team is actively using the existing W&B Model Registry to organize your models, this will still be available through the new Registry UI. Navigate to the Model Registry from the homepage, and the banner will allow to select a team and visit it's model registry.

https://github.com/wandb/docodile/assets/40642416/3b4376ee-0464-45ee-b5dd-46f6b0b59dcf

Please look out for incoming information on a migration we will be making available to migrate contents from the current model registry into the new model registry inside W&B Registry. You can reach out to support@wandb.com with any questions or to speak to our product team about any concerns with the migration.
```

---
slug: /guides/registry
displayed_sidebar: default
---

> import Translate, {translate} from '@docusaurus/Translate';

Author: Will we have a live asset that customers can see?

Reviewer: at the moment we won't have a fully public registries app the same way we do for projects. Users need to be in a given org to see Registries (and be a W&B user). If the registry visibility is Organization - to see the wandb registry a user must be in the wandb team. 

But I will let you know if/when we add a fully public registry that can become a demo registry

import { CTAButtons } from '@site/src/components/CTAButtons/CTAButtons.tsx';

> # Registry

Author: @noaleetz This is the page that will probably need the most review; how to capture user's attention, tell them the why, what, and how + next steps.

## How it works
W&B Registry is composed of three main components: a registry, collection, and [artifact versions](../artifacts/create-a-new-artifact-version.md).

A *registry* is a [INSERT]. You can think of a registry as the top most level of a directory. Each registry consists of one or more sub directories called collections. A *collection* is a folder or a set of linked [artifact versions](../artifacts/create-a-new-artifact-version.md) inside a registry.

Consider the following example, suppose your company wants to explore brand new cat classification models within your organization. To do this, you create a registry called "Cat Models registry". You are not sure which image classification is best, so you create multiple machine learning experiments with varying hyperparameters and algorithms. To organize your experiments and results, you create a collection for each model algorithm that experiment. Within that collection, you link the best model artifacts from your experiments to that collection.

<!-- To do: Add Raven's new diagram -->

## How to get started
Depending on your use case, explore the following resources to get started with the W&B Registry:

> <!-- To do: INSERT -->

Author: Besides links to the registry dev guide, are there other resources that we want to link to here? If yes, what?

Reviewer: Can we make sure to be consistent with the naming of the product - W&B Registries?

Reviewer: For now we don't have any content but hoping we'll get some good tutorial videos the closer we get. here is the coming soon article - https://wandb.ai/wandb_fc/product-announcements-fc/reports/Coming-soon-Registries-in-W-B-Models--Vmlldzo3OTc4OTg1
